### PR TITLE
improvement: Support numbers with commas in them

### PR DESCRIPTION
**Pull request by vjeux**
I tried to paste weekly active users from excel but isNumeric didn't recognize "10,000" as number and wouldn't recognize the data as a chart. This PR fixes it.

### CODE REVIEW

File: src/analytics.ts [hunk 1]
```diff
@@ -11,14 +11,17 @@ export const EVENT_SHAPE = "shape";
...
+        window.gtag("event", name, {
+          event_category: category,
+          event_label: label,
+          value,
+        });
+      }
+    : typeof process !== "undefined" && process?.env?.JEST_WORKER_ID
+    ? (category: string, name: string, label?: string, value?: number) => {}
+    : (category: string, name: string, label?: string, value?: number) => {
+        console.info("Track Event", category, name, label, value);
```
Comment (vjeux) [inline src/analytics.ts:26]: This was spamming the console during tests. I just mock it in the test environment

File: src/charts.ts [hunk 2]
```diff
@@ -19,15 +19,15 @@ export const NOT_SPREADSHEET = "NOT_SPREADSHEET";
 export const VALID_SPREADSHEET = "VALID_SPREADSHEET";
 
 type ParseSpreadsheetResult =
-  | { type: typeof NOT_SPREADSHEET }
+  | { type: typeof NOT_SPREADSHEET; reason: string }
```
Comment (vjeux) [inline src/charts.ts:22]: I'm adding a reason so it's easier to debug why it didn't parse.

File: src/charts.ts [hunk 3]
```diff
@@ -19,15 +19,15 @@ export const NOT_SPREADSHEET = "NOT_SPREADSHEET";
...
   | { type: typeof VALID_SPREADSHEET; spreadsheet: Spreadsheet };
 
 const tryParseNumber = (s: string): number | null => {
-  const match = /^[$€£¥₩]?([0-9]+(\.[0-9]+)?)$/.exec(s);
+  const match = /^[$€£¥₩]?([0-9,]+(\.[0-9]+)?)$/.exec(s);
```
Comment (vjeux) [inline src/charts.ts:26]: This is the fix [1/2]

File: src/charts.ts [hunk 4]
```diff
@@ -19,15 +19,15 @@ export const NOT_SPREADSHEET = "NOT_SPREADSHEET";
...
   if (!match) {
     return null;
   }
-  return parseFloat(match[1]);
+  return parseFloat(match[1].replace(/,/g, ""));
```
Comment (vjeux) [inline src/charts.ts:30]: This is the fix [2/2]

File: src/charts.ts [hunk 5]
```diff
@@ -104,13 +104,13 @@ export const tryParseSpreadsheet = (text: string): ParseSpreadsheetResult => {
...
 
-  // Check for tab separeted values
+  // Check for tab separated values
   let lines = text
     .trim()
     .split("\n")
     .map((line) => line.trim().split("\t"));
 
-  // Check for comma separeted files
+  // Check for comma separated files
```
Comment (vjeux) [inline src/charts.ts:113]: typo

File: src/tests/__snapshots__/charts.test.tsx.snap [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      "Week 3",
+    ],
+    "title": "Users",
+    "values": Array [
+      814,
+      10301,
+      4264,
+    ],
+  },
+  "type": "VALID_SPREADSHEET",
```
Comment (vjeux) [inline src/tests/__snapshots__/charts.test.tsx.snap:18]: Now it parses it correctly!

File: src/tests/charts.test.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+describe("tryParseSpreadsheet", () => {
+  it("works for numbers with comma in them", () => {
+    const result = tryParseSpreadsheet(
+      `Week Index${"\t"}Users
+Week 1${"\t"}814
+Week 2${"\t"}10,301
+Week 3${"\t"}4,264`,
+    );
+    expect(result).toMatchSnapshot();
```
Comment (vjeux) [inline src/tests/charts.test.tsx:11]: @lipis here, super easy to add a test for parsing with snapshot testing :)

Comment (lipis) [reply src/tests/charts.test.tsx:11]: Thanks @vjeux!

**Review by lipis — APPROVED** [review]

**Pull request merged**